### PR TITLE
Re-add foxy builds to main branch

### PR DESCRIPTION
**Pull request by fmauch**
Obviously I was not aware that workflow definitions have to reside on the main branch...

### CODE REVIEW

**Comment by fmauch** [comment]
Actually, I am not sure whether this is correct...

File: .github/workflows/foxy-binary-build.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: Foxy Binary Build
+on:
+  push:
+    branches:
```
Comment (t-schnell) [inline .github/workflows/foxy-binary-build.yml:4]: Maybe remove this here and only leave it on the foxy branch - similar to the galactic solution?

File: .github/workflows/foxy-semi-binary-build.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+name: Foxy Semi-Binary Build
+on:
+  push:
+    branches:
```
Comment (t-schnell) [inline .github/workflows/foxy-semi-binary-build.yml:4]: as above

File: .github/workflows/foxy-source-build.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+name: Foxy Source Build
+on:
+  push:
+    branches:
```
Comment (t-schnell) [inline .github/workflows/foxy-source-build.yml:4]: as above

**Review by t-schnell — APPROVED** [review]
As written at the point in the review - we maybe should remove the on push action of the branch in main, and only leave the scheduled actions to be consistent to the way it is handled with galactic (unless I'm missing something). As this is not critical, I will approve either way.

**Pull request merged**